patch-upstream.sh

```diff
@@ -27,5 +27,4 @@ go mod edit -replace github.com/open-telemetry/opentelemetry-collector-contrib/i
 
 # A simple `go mod tidy` does not work.
 # See: https://github.com/aws-observability/aws-otel-collector/issues/926
-rm -fr go.sum
-go mod tidy -compat=1.18
+rm go.sum && go mod tidy -go=1.16 && go mod tidy -go=1.18
```

Comment:
I am curious why this could still be required in our distro. I checked both collector contrib and core and it is not required there. Are we missing something?

Reply:
As i am understanding this after we replace with our ADOT components, we need to download those packages to be included in build.

Reply:
Sorry, I meant why is `go=1.16` and `go=1.18` needed rather just a standard `go mod tidy`. I believe collector core runs `go mod tidy -compat=1.18` for all of it's packages. Same with [contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7677ba717f419405d8fca088f88de5c2eb438df3/Makefile.Common#L118).

Reply:
I don't think we need to maintain 1.16 compatibility anymore, so we should simply be able to use a single `go mod tidy -compat=<min_supported_version>`.  I think at this time `min_supported_version` == `1.18`.

Reply:
@vasireddy99 try running `go clean -modcache` before running `go mod tidy -compat=1.18`